### PR TITLE
storage: Fix ttl expire ts overflow

### PR DESCRIPTION
**Pull request by Connor1996**
### What problem does this PR solve?

When ttl is set to a very large value, such u64::MAX, calculating expire ts(current_ts + ttl) would overflow. 

### What is changed and how it works?

using `saturating_add` for calculating ttl expire ts.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

- None

### CODE REVIEW

File: components/engine_traits/src/util.rs [hunk 1]
```diff
@@ -4,6 +4,8 @@ use super::{Error, Result};
 use tikv_util::codec;
 use tikv_util::codec::number::{self, NumberEncoder};
 
+pub const TEST_CURRENT_TS: u64 = 100;
```
Comment (BusyJay) [inline components/engine_traits/src/util.rs:7]: Why define it twice?

Comment (Connor1996) [reply components/engine_traits/src/util.rs]: forget to clean up

**Comment by Connor1996** [comment]
PTAL again

**Review by BusyJay — APPROVED** [review]

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**